assets/js/modules/analytics-4/utils/extract-dashboard-data.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * Analytics-4 Dashboard data extraction functions.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { each } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getLocale } from '../../../util/i18n';
+import {
+	convertSecondsToArray,
+	numFmt,
+	getChartDifferenceArrow,
+	calculateDifferenceBetweenChartValues,
+} from '../../../util';
+import { partitionReport } from '../../../util/partition-report';
+import parseDimensionStringToDate from '../../analytics/util/parseDimensionStringToDate';
+
+/**
+ * Reduces and processes an array of analytics row data.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array}  rows                 An array of rows to reduce.
+ * @param {number} selectedMetricsIndex The index of metrics array in the metrics set.
+ * @param {number} selectedStats        The currently selected stat we need to return data for.
+ * @return {Array} Array of selected stats from analytics row data.
+ */
+function reduceAnalyticsRowsData( rows, selectedMetricsIndex, selectedStats ) {
+	const dataMap = [];
+	each( rows, ( row ) => {
+		if ( row.metricValues ) {
```

Comment:
Probably better to return early to reduce nesting:

```js
if ( ! rows.metricValues ) {
    return;
}

const { value } = ...
```

Reply:
Updated 👍